src/components/AuthForm.js

```diff
@@ -0,0 +1,127 @@
+import {
+  Form,
+  Link,
+  useSearchParams,
+  useActionData,
+  useNavigation,
+  useNavigate,
+} from "react-router-dom";
+
+import classes from "./AuthForm.module.css";
+import { useState } from "react";
+import { fetch_url } from "../services";
+import { toast } from "react-toastify";
+
+function AuthForm() {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const isLogin = searchParams.get("mode") === "login";
+  const [otp, setOtp] = useState(false);
+  const navigation = useNavigation();
+  const isSubmitting = navigation.state === "submitting";
+  const data = useActionData();
+  const navigate = useNavigate();
+
+  async function submitHandler(event) {
+    event.preventDefault();
+    if (!otp) {
+      const data = new FormData(event.target);
+      var object = {};
+      data.forEach((value, key) => (object[key] = value));
+      var body = JSON.stringify(object);
+
+      let url = process.env.REACT_APP_BACKEND_URL;
+      url += isLogin ? "user/login/" : "user/register_user/";
+
+      const [ok, output] = await fetch_url(url, body, "POST");
+
+      if (!ok) {
+        return toast.error("Something bad happened");
+      }
+
+      if (isLogin) {
+        if (output.status === "GA" || output.status === "Email") {
+          setOtp(true);
+        } else {
+          let jwt_token = output.status.access;
+          localStorage.setItem("token", jwt_token);
+          toast.success("Successfully Logged In");
+          return navigate("/");
+        }
+      } else {
+        return navigate("/auth?mode=login");
+      }
+    } else {
+      const data = new FormData(event.target);
+      object = {};
+      data.forEach((value, key) => (object[key] = value));
+
+      let body = JSON.stringify(object);
+      let url = `${process.env.REACT_APP_BACKEND_URL}/user/check_otp/`;
+
+      let [ok, output] = await fetch_url(url, body, "POST");
+
+      if (ok) {
+        let jwt_token = output.status.access;
```

Comment:
Can we add these three lines to a method and except writing it again call this method?